server/src/test/scala/org/http4s/server/middleware/LoggerSpec.scala

```diff
@@ -0,0 +1,64 @@
+package org.http4s
+package server
+package middleware
+
+import org.http4s.dsl._
+
+/**
+  * Common Tests for Logger, RequestLogger, and ResponseLogger
+  */
+class LoggerSpec extends Http4sSpec {
+
+  val testService = HttpService {
+    case req @ GET -> Root / "request" =>
+      Ok("request response")
+    case req @ POST -> Root / "post" =>
+      Ok("post response")
+  }
+
+  val urlForm = UrlForm("foo" -> "bar")
+
+  "ResponseLogger" should {
+    val responseLoggerService = ResponseLogger(true, true)(testService)
+
+    "not effect a Get" in {
+      val req = Request(uri = uri("/request"))
+      responseLoggerService.orNotFound(req) must returnStatus(Status.Ok)
+    }
+
+    "not effect a Post" in {
+      val req = Request(uri = uri("/post"), method = POST).withBody(urlForm)
+      req.flatMap(responseLoggerService.orNotFound) must returnStatus(Status.Ok)
+    }
+  }
+
+  "RequestLogger" should {
+    val requestLoggerService = RequestLogger(true, true)(testService)
+
+    "not effect a Get" in {
+      val req = Request(uri = uri("/request"))
+      requestLoggerService.orNotFound(req) must returnStatus(Status.Ok)
+    }
+
+    "not effect a Post" in {
+      val req = Request(uri = uri("/post"), method = POST).withBody(urlForm)
```

Comment:
Not yet, needs a test for that. Working on that one. The fact that it now buffers it and then advances it though makes me more optimistic towards it's success in this regard.